opsmngr/logs.go

```diff
@@ -0,0 +1,272 @@
+	Get(context.Context, string, string, *LogListOptions) (*Log, *atlas.Response, error)
+	Retry(context.Context, string, string) (*atlas.Response, error)
+	Download(context.Context, string, string, io.Writer) (*atlas.Response, error)
+	Create(context.Context, string, *Log) (*Log, *atlas.Response, error)
+	Extend(context.Context, string, string, *Log) (*atlas.Response, error)
+	Delete(context.Context, string, string) (*atlas.Response, error)
+}
+
+// LogsServiceOp handles communication with the Log Collection Jobs related methods of the
+// MongoDB Ops Manager API
+type LogsServiceOp struct {
+	Client RequestDoer
+}
+
+var _ LogsService = &LogsServiceOp{}
+
+// Log represents a Log Collection Job in the MongoDB Ops Manager API.
+type Log struct {
+	ID                         string      `json:"id,omitempty"`
+	GroupID                    string      `json:"groupId,omitempty"`
+	UserID                     string      `json:"userId,omitempty"`
+	CreationDate               string      `json:"creationDate,omitempty"`
+	ExpirationDate             string      `json:"expirationDate,omitempty"`
+	Status                     string      `json:"status,omitempty"`
+	ResourceType               string      `json:"resourceType,omitempty"`
+	ResourceName               string      `json:"resourceName,omitempty"`
+	RootResourceName           string      `json:"rootResourceName,omitempty"`
+	RootResourceType           string      `json:"rootResourceType,omitempty"`
+	URL                        string      `json:"downloadUrl,omitempty"`
+	Redacted                   bool        `json:"redacted,omitempty"`
+	LogTypes                   []string    `json:"logTypes,omitempty"`
+	SizeRequestedPerFileBytes  int64       `json:"sizeRequestedPerFileBytes,omitempty"`
+	UncompressedDiskSpaceBytes int64       `json:"uncompressedSizeTotalBytes,omitempty"`
+	ChildJob                   []*ChildJob `json:"childJobs,omitempty"` //included if verbose is true
+}
+
+// ChildJob represents a ChildJob in the MongoDB Ops Manager API.
+type ChildJob struct {
+	AutomationAgentID          string `json:"automationAgentId,omitempty"`
+	ErrorMessage               string `json:"errorMessage,omitempty"`
+	FinishDate                 string `json:"finishDate"`
+	HostName                   string `json:"hostName"`
+	LogCollectionType          string `json:"logCollectionType"`
+	Path                       string `json:"path"`
+	StartDate                  string `json:"startDate"`
+	Status                     string `json:"status"`
+	UncompressedDiskSpaceBytes int64  `json:"uncompressedDiskSpaceBytes"`
+}
+
+// Logs represents a array of Logs
+type Logs struct {
+	Links      []*atlas.Link `json:"links"`
+	Results    []*Log        `json:"results"`
+	TotalCount int           `json:"totalCount"`
+}
+
+// LogListOptions specifies the optional parameters to List methods that
+// support pagination.
+type LogListOptions struct {
+	atlas.ListOptions
+	Verbose bool `url:"verbose,omitempty"`
```

Comment:
what's the default according to docs? if a user does not pass a value does it default to `verbose =true` or `verbose=false`?
if it defaults to `verbose=false` this can stay as it is
if it defaults to `verbose=true` you may need to remove `omitempty` as it would not be serialized and then fallback to true

Reply:
it is `false`